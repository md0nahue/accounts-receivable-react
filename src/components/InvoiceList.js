import React from "react";
import { useSelector } from "react-redux";
import { NavLink } from "react-router-dom";

export function InvoiceList() {
  const invoices = useSelector((state) => state.invoices);

  return (
    <ul className="InvoiceList nav nav-pills flex-column">
      {invoices.map((invoice) => (
        <li className="nav-item" key={invoice.id}>
          <NavLink to={`/invoices/${invoice.id}`} className="nav-link">
            #{invoice.id} {invoice.client}
          </NavLink>
        </li>
      ))}
    </ul>
  );
}
