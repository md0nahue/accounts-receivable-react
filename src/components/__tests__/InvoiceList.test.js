import React from "react";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { render, cleanup } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import "jest-dom/extend-expect";

import InvoiceList from "../InvoiceList";

const middlewares = [];
const mockStore = configureStore(middlewares);

beforeEach(cleanup);

describe("InvoiceList Component", () => {
  it("Renders a list of provided invoices", () => {
    const invoices = [
      { id: 1, client: "Test Double 1" },
      { id: 2, client: "Test Double 2" },
      { id: 3, client: "Test Double 3" },
    ];
    const store = mockStore({ invoices });

    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <InvoiceList />
        </MemoryRouter>
      </Provider>
    );

    const invoiceList = container.querySelector(".InvoiceList");
    expect(invoiceList.children.length).toEqual(invoices.length);
    expect(container.firstChild).toMatchSnapshot();
  });
});
