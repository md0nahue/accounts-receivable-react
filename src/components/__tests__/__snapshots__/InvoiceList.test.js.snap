// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoiceList Renders a list of provided invoices 1`] = `
<ul
  class="InvoiceList nav nav-pills flex-column"
>
  <li
    class="nav-item"
  >
    <a
      class="nav-link"
      href="/invoices/1"
    >
      #
      1
       
      Test Double 1
    </a>
  </li>
  <li
    class="nav-item"
  >
    <a
      class="nav-link"
      href="/invoices/2"
    >
      #
      2
       
      Test Double 2
    </a>
  </li>
  <li
    class="nav-item"
  >
    <a
      class="nav-link"
      href="/invoices/3"
    >
      #
      3
       
      Test Double 3
    </a>
  </li>
</ul>
`;
